shell/platform/android/io/flutter/app/FlutterPluginRegistry.java

```diff
@@ -43,8 +47,10 @@
     private final List<UserLeaveHintListener> mUserLeaveHintListeners = new ArrayList<>(0);
     private final List<ViewDestroyListener> mViewDestroyListeners = new ArrayList<>(0);
 
-    public FlutterPluginRegistry(FlutterNativeView nativeView, Context context) {
+    // NOTE: Either nativeView needs to be null, XOR flutterEngine needs to be null
```

Comment:
Would this be better expressed as a runtime check + exception?

Reply:
I opted instead to break the constructor into 2 versions: old vs new.  Let me know if that sounds problematic for any reason.